modules/controller/dh-controller.js

```diff
@@ -194,6 +197,93 @@ class DHController {
         }
     }
 
+    async lookupTrail(req, res) {
+        this.logger.api('POST: Trail lookup request received.');
+
+        if (req.body === undefined ||
+            req.body.identifier_types === undefined ||
+            req.body.identifier_values === undefined
+        ) {
+            res.status(400);
+            res.send({
+                message: 'Bad request',
+            });
+            return;
+        }
+
+        const { identifier_types, identifier_values, opcode } = req.body;
+
+
+        try {
+            const response = await this.trailService.lookupTrail(
+                identifier_types,
+                identifier_values,
+                opcode,
+            );
+
+            res.status(200);
+            res.send(response);
+        } catch (e) {
+            res.status(400);
+            res.send(e);
```

Comment:
If this is exception that we did not expect, we should return 501 to user. Also we should not return res.send(e) we should return our message to user formatted something like:
{
errorMessage: 'Internal server error';
}